model_predictive_speed_and_steer_control.py

```
@@ -0,0 +1,775 @@
+
+def iterative_linear_mpc_control(xref, x0, dref, oa, od):
+    """
+    MPC control with updating operational point iteratively
+    """
+    ox, oy, oyaw, ov = None, None, None, None
+
+    if oa is None or od is None:
+        oa = [0.0] * T
+        od = [0.0] * T
+
+    for i in range(MAX_ITER):
```

Ruff (0.14.5)

Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)